tests/CommonExtractor.php

```diff
@@ -0,0 +1,497 @@
+                "nullable" => ($column['IS_NULLABLE'] === "NO") ? false : true,
+                "default" => $column['COLUMN_DEFAULT'],
+                "ordinalPosition" => $column['ORDINAL_POSITION'],
+            ];
+
+            if (array_key_exists('CONSTRAINT_NAME', $column) && !is_null($column['CONSTRAINT_NAME'])) {
+                $curColumn['constraintName'] = $column['CONSTRAINT_NAME'];
+            }
+            if (array_key_exists('REFERENCED_TABLE_NAME', $column) && !is_null($column['REFERENCED_TABLE_NAME'])) {
+                $curColumn['foreignKeyRefSchema'] = $column['REFERENCED_TABLE_SCHEMA'];
+                $curColumn['foreignKeyRefTable'] = $column['REFERENCED_TABLE_NAME'];
+                $curColumn['foreignKeyRefColumn'] = $column['REFERENCED_COLUMN_NAME'];
+            }
+            if ($column['EXTRA']) {
+                $curColumn["extra"] = $column["EXTRA"];
+                if ($column['EXTRA'] === 'auto_increment') {
+                    $curColumn['autoIncrement'] = $tableDefs[$curTable]['autoIncrement'];
+                }
+                if ($column['EXTRA'] === 'on update CURRENT_TIMESTAMP'
+                    && $column['COLUMN_DEFAULT'] === 'CURRENT_TIMESTAMP') {
+                    $tableDefs[$curTable]['timestampUpdateColumn'] = $column['COLUMN_NAME'];
+                }
+            }
+            $tableDefs[$curTable]['columns'][$column['ORDINAL_POSITION'] - 1] = $curColumn;
+        }
+        return array_values($tableDefs);
+    }
+
+    public function testConnection(): void
+    {
+        try {
+            $connection = $this->getConnection();
+
+            /** @var \PDOStatement $stmt */
+            $stmt = $connection->query('SELECT 1');
+            $stmt->execute();
+        } catch (\Throwable $exception) {
+            throw new UserException($exception->getMessage());
+        }
+    }
+
+    public function validateIncrementalFetching(
+        TableDetailParameters $table,
+        string $columnName,
+        ?int $limit = null
+    ): void {
+        $db = $this->getConnection();
+        /** @var \PDOStatement $res */
+        $res = $db->query(
+            sprintf(
+                'SELECT * FROM INFORMATION_SCHEMA.COLUMNS as cols 
+                            WHERE TABLE_SCHEMA = %s AND TABLE_NAME = %s AND COLUMN_NAME = %s',
+                $db->quote($table->getSchema()),
+                $db->quote($table->getTableName()),
+                $db->quote($columnName)
+            )
+        );
+
+        /** @var array $columns */
+        $columns = $res->fetchAll();
+        if (count($columns) === 0) {
+            throw new UserException(
+                sprintf(
+                    'Column [%s] specified for incremental fetching was not found in the table',
+                    $columnName
+                )
+            );
+        }
+        if ($columns[0]['EXTRA'] === 'auto_increment') {
+            $this->incrementalFetching['column'] = $columnName;
```

Comment:
Jak jsi přišel na to, že to má vypadat zrovna takhle? Tos musel někde opsat/vykoukat z DB? Nemělo by to teda spíš vracet zas value object `IncrementalFetchingSettings` nebo tak něco? A ten by byl pak k dispozici v extractoru?

Reply:
Nebo je to úplně DB-dependent co v tom je? Pak by zas nedávalo smysl mít to validateIncrementalFetching, ne?

Reply:
Dalsi historicky magic array, zalozim issue.